src/libopensc/errors.h

```diff
@@ -130,7 +130,7 @@ extern "C" {
 #define SC_ERROR_UNKNOWN			-1900
 #define SC_ERROR_PKCS15_APP_NOT_FOUND		-1901
 
-const char *sc_strerror(int sc_errno);
+const char *sc_strerror(long int sc_errno);
```

Comment:
I dont think this is the correct solution. The problem is really on the line pointed out by the oss-fuzz, because this is the only value that can not be inverted in the `int` type:

https://github.com/OpenSC/OpenSC/blob/a6f363b92963b635d14897c9fef18c5354be4ff5/src/libopensc/errors.c#L157

I think the function should not try to change the value in `error` from arguments, but just store it in temporary `unsigned int` variable, as from that line on, it is always positive value.

Reply:
Indeed, I've just over overseen that, thanks